### PR TITLE
MongoDB Interface for the Bot.

### PR DESCRIPTION
**Pull request by Rahtoken**
This PR will track the progress for the migration and interfacing of the storage backend of the Bot from NetDB to MongoDB.

### CODE REVIEW

File: database.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+const mongoose = require('mongoose');
+const User = require('./models/User');
+
+let connect = (dbUrl) => {
```
Comment (Aditya-Kolla) [inline database.js]: const instead of let would be better

Comment (Rahtoken) [reply database.js]: Noted, I will make it const in the next iteration.

File: models/User.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+const mongoose = require('mongoose');
+
+let UserSchema = new mongoose.Schema({
```
Comment (Rahtoken) [inline models/User.js]: @Aditya-Kolla have a look at the Schema for the user.

File: database.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+/**
+ * Add a new player to the database.
+ * @param {*} discordId The Discord ID of the new player.
+ * @param {*} openDotaId The OpenDota ID of the new player.
+ * @returns The newly added player, null if the player already exists.
+ */
+const addPlayer = (discordId, openDotaId) => {
+    return getPlayer(discordId)
+        .then(player => {
+            if (player != null) {
```
[the remark this answers is not in the document]
Comment (Rahtoken) [reply database.js]: Noted.
I will take care of reducing code clutter next PR.

**Pull request merged**